packages/sol-doc/src/solidity_doc_generator.ts

```diff
@@ -26,6 +28,19 @@ import {
     TypeDocTypes,
 } from '@0xproject/types';
 
+// Unforunately, the only way to currently retrieve the declared structs with Solidity contracts
+// is to tease it out of the params/return values included in it's ABI. These structures do
+// not include the structs actual name, so we needed a mapping to assign the proper name to a
+// struct. If the name is not in this mapping, the structs name will default to the param/return value
+// name.
+const customTypeHashToName: { [hash: string]: string } = {
+    '52d4a768701076c7bac06e386e430883975eb398732eccba797fd09dd064a60e': 'Order',
+    '46f7e8c4d144d11a72ce5338458ea37b933500d7a65e740cbca6d16e350eaa48': 'FillResult',
+    c22239cf0d29df1e6cf1be54f21692a8c0b3a48b9367540d4ffff4608b331ce9: 'OrderInfo',
+    c21e9ff31a30941c22e1cb43752114bb467c34dea58947f98966c9030fc8e4a9: 'TraderInfo',
+    '07c2bddc165e0b5005e6244dd4a9771fa61c78c4f42abd687d57567b0768136c': 'MatchedFillResult',
+};
```

Comment:
Since this was previously in the global state, but now is being passed in, I've exported a class rather then a function, so as to have state accessible across functions without having to pass it down 5 levels of nested functions.